solutions/kattis-pointinpolygon.mdx

```diff
@@ -0,0 +1,147 @@
+---
+
+**Time Complexity:** $\mathcal O(NM)$
+## Solution 
+In order to determine if a point is inside a polygon, we can imagine a ray being casted from that point towards any direction. Resulting in three possible scenarios: 
+
+1. If the ray intersects with odd number of polygon edges, then the point is inside.
+2. If the ray intersects with even number of polygon edges, then the point is outside.
+3. If the origin point is on a line segment, then the point is on the edge.
+
+This algorithm is known as the [*Ray Casting Algorithm*](https://en.wikipedia.org/wiki/Point_in_polygon#Ray_casting_algorithm), more details can be found [here](https://en.wikipedia.org/wiki/Point_in_polygon)
+
+## Line segment & Line segment intersection
+
+Given two line segments $L_1 = (p_1,p_2)$ and $L_2 = (p_3,p_4)$, how can we effectively check if they intersect each other? We will first have to use a concept from linear algebra called the *Determinant*.
+
+<Info title="Complementary resource">
+
+If you don't already know what a determinant is, watch this [youtube video](https://www.youtube.com/watch?v=Ip3X9LOh2dk) for an in-depth explanation.
+
+</Info>
+
+We will define the determinant operator of two vectors as $\times$. Given two vectors $\vec{a}=\langle x_1,y_1 \rangle$ and $\vec{b}=\langle x_2,y_2 \rangle$. If $\vec{a}\times\vec{b}$ gives a positive area then $\vec{b}$ points to the left of $\vec{a}$, if it gives a negative area then $\vec{b}$ points to the right of $\vec{a}$, and if it gives a zero area then $\vec{b}$ and $\vec{a}$ are parallel.
+
+To check if $L_1$ and $L_2$ intersect with each other, is equal to checking if $L_2$ intersects with the line from line segment $L_1$, and checking if $L_1$ intersects with the line from line segment $L_2$. For the first of two condition to be true, $p_3$ and $p_4$ from $L_2$ must be on the opposite directions of $L_1$. To define this mathematically, let
+	
+$$
+\vec{a} = p_2-p_1
+$$
+$$
+\vec{b_1} = p_3-p_2
+$$
+$$
+\vec{b_2} = p_4-p_2
+$$
+
+The condition will be true if and only if $\vec{a}\times\vec{b_1} \neq \vec{a}\times\vec{b_2}$. Verifying the second of the two cases is equivalent to the procedure above, except $p_1$ and $p_2$ are switched with $p_3$ and $p_4$.
+
+As all geometry problem go, edge cases are a must. One edge case one might consider in our method above is that if both $\vec{a}\times\vec{b_1}$ and $\vec{a}\times\vec{b_2}$ are equal and are equal to 0, then they can still intersect as long as they overlap with each other. But let me in you a ***secret***, due to the way this problem is structured, all vertices are on lattice points, meaning that they have whole numbers as coordinates. And since one of the $L$ is customizable (we get to pick the point the point cast its ray at), we can just set its coordinate to $(\infty,1)$. This way, the ray will have a slope of close to $0$, successfully avoiding all possible lattice point intersections. One trick to rule edges cases all! 
```

Comment:
> but let me in you a secret

did you mean but let me let you in on a secret?

Reply:
oops, you are right! Thanks for catching the mistake haha.